src/main.cpp

```diff
@@ -26,14 +26,37 @@ bool isInRecoveryMode() {
   }
   saveSettings();
   DPRINTF("Wrote QPC = %d\n", Settings->quick_power_cycle_count);
+  // making sure the board can't go into a locked state
+  Settings->board_locked = false;
   return recovery;
+  return false;
+}
+
+bool isLocked() {
+  // no password set or board isn't armed, won't lock
+  if (strlen(Settings->ap_self_password) == 0 || !Settings->board_lock_armed ||
+      !Settings->board_locked) {
+    bool recovery = isInRecoveryMode(); // recovery mode check
+    Settings->board_locked = false;
+    return recovery; // passing recovery mode
+  } else if (Settings->quick_power_cycle_count >
+             0) { // password is set and board is armed
+    Settings->quick_power_cycle_count = 0;
+    Settings->board_locked = true;
+  } else {
+    Settings->quick_power_cycle_count++;
+    // Reset in 10 seconds;
+    TaskQueue.postOneShotTask(resetQuickPowerCycleCount, 10000L);
+  }
+  saveSettings();
+  return false; // not in recovery mode
 }
 
 extern "C" void setup() {
   WiFi.persistent(false);
   loadSettings();
 
-  if (isInRecoveryMode()) {
+  if (isLocked()) {
```

Comment:
After my suggestions this would turn into:
```
manageLockedStateTransitionOnStartup();
if (!Settings->board_locked && isInRecoveryMode()) {
  recovery_setup();
}
```